src/action/macos/create_fstab_entry.rs

```diff
@@ -236,76 +148,67 @@ impl Action for CreateFstabEntry {
+        let mut fstab_buf = String::new();
+        fstab
+            .read_to_string(&mut fstab_buf)
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Read(fstab_path.to_owned(), e)))?;
 
-            if let Some(start) = file_contents.rfind(fstab_entry.as_str()) {
-                let end = start + fstab_entry.len();
-                file_contents.replace_range(start..end, "")
-            }
+        let updated_buf = fstab_buf
+            .lines()
+            .filter_map(|line| {
+                // Delete nix-installer entries with a "prelude" comment
+                if line.starts_with("# nix-installer") {
+                    None
+                } else {
+                    Some(line)
+                }
+            })
+            .filter_map(|line| {
+                if line.split(&[' ', '\t']).nth(2) == Some("/nix") {
+                    // Delete the mount line for /nix
+                    None
+                } else {
+                    Some(line)
+                }
+            })
+            .collect::<Vec<&str>>()
+            .join("\n");
 
-            file.seek(SeekFrom::Start(0))
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Seek(fstab_path.to_owned(), e)))?;
-            file.set_len(0)
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Truncate(fstab_path.to_owned(), e)))?;
-            file.write_all(file_contents.as_bytes())
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Write(fstab_path.to_owned(), e)))?;
-            file.flush()
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Flush(fstab_path.to_owned(), e)))?;
-        } else {
-            return Err(Self::error(CreateFstabEntryError::CannotDetermineFstabLine));
-        }
+        fstab
+            .seek(SeekFrom::Start(0))
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Seek(fstab_path.to_owned(), e)))?;
+        fstab
+            .set_len(0)
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Truncate(fstab_path.to_owned(), e)))?;
+        fstab
+            .write_all(updated_buf.as_bytes())
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Write(fstab_path.to_owned(), e)))?;
 
         Ok(())
     }
 }
 
-fn fstab_lines(uuid: &Uuid, apfs_volume_label: &str) -> String {
-    let prelude_comment = fstab_prelude_comment(apfs_volume_label);
-    let fstab_entry = fstab_entry(uuid);
-    prelude_comment + "\n" + &fstab_entry
-}
-
-fn fstab_prelude_comment(apfs_volume_label: &str) -> String {
-    format!("# nix-installer created volume labelled `{apfs_volume_label}`")
-}
-
 fn fstab_entry(uuid: &Uuid) -> String {
-    format!("UUID={uuid} /nix apfs rw,noauto,nobrowse,suid,owners")
+    format!("UUID={uuid} /nix apfs rw,noatime,noauto,nobrowse,nosuid,owners # Added by the Determinate Nix Installer")
```

Comment:
This now differs a lot more from how upstream configures their volume:

https://github.com/NixOS/nix/blob/a8a572b11b3f29cdb2f57a94e32cf72a5ecbf6dc/scripts/create-darwin-volume.sh#L466

These should go in a separate commit with rationale behind the change (other than "I wanted to"....), please.

Reply:
https://github.com/DeterminateSystems/nix-installer/pull/1338/commits/7bb01781cb7b676e286883f5b56fc1e22f91b054 I think I did that?

Reply:
Updated the commit text!